### PR TITLE
On merge/update error, don't retry the same PR

### PR DESCRIPTION
**Pull request by sileht**
On merge/update error, don't retry the same PR
Capture some exceptions with sentry

### CODE REVIEW

**Comment by sileht** [comment]
Pull-request updated, HEAD is now 47f89a5c5f18e9e45dd735e7b7ef343bf4677c15

**Comment by sileht** [comment]
Pull-request updated, HEAD is now 76e0bff4423d5bceb50538f35bed2d96c9fbad4b

**Comment by sileht** [comment]
Pull-request updated, HEAD is now 543659c424bbcc032a0b6190d0cff391b24d0076

File: mergify_engine/backports.py [hunk 1]
```diff
@@ -139,6 +139,7 @@ def _backport(repo, pull, branch_name, installation_token):
 
         git("push", "origin", bp_branch)
     except Exception:
+        utils.get_sentry().captureException()
```
Comment (jd) [inline mergify_engine/backports.py]: I'd rather use sentry as a log handler
https://docs.sentry.io/clients/python/integrations/logging/#usage

File: mergify_engine/engine.py [hunk 2]
```diff
@@ -405,6 +406,7 @@ def _get_next_pull_to_processed(self, branch, **context):
             else:
                 return p
 
+    @tenacity.retry(retry=tenacity.retry_if_exception_type(tenacity.TryAgain))
```
Comment (jd) [inline mergify_engine/engine.py]: just user `retry_never`

File: mergify_engine/backports.py (hunk 1 above)
Comment (sileht) [reply mergify_engine/backports.py]: done

File: mergify_engine/engine.py (hunk 2 above)
Comment (sileht) [reply mergify_engine/engine.py]: done

**Comment by sileht** [comment]
Pull-request updated, HEAD is now cb37994ce2264ff30ca86514b46519f8a79d6c4f

**Comment by sileht** [comment]
Pull-request updated, HEAD is now 63921a92b8aefcd90577cbc39dc916b590d6eb6c

**Comment by sileht** [comment]
Pull-request updated, HEAD is now 4fb9bfbdf44ab2b42d5de555878d5c34e561c0ff

**Pull request merged**